src/ast.fs

```diff
@@ -59,7 +59,7 @@ and TypeSpec =
 
 and Type = {
     name: TypeSpec // e.g. int
-    typeQ: string option // e.g. const, uniform
+    typeQ: string list option // e.g. const, uniform
```

Comment:
If you use simply `string list`, that should simplify the code.

Reply:
Indeed, good idea, thanks! Will update later.